### ISSUE TITLE
Add cm support for measure input

### ISSUE THREAD

**Issue by AndreeaCristinaRadacina** [Codeinwp/woocommerce-product-addon#239]
### What problem does this address?

A user asked about the possibility of having cm instead of m in the measure input.

ref: https://secure.helpscout.net/conversation/2484497986/401016?viewId=212385

### What is your proposed solution?

Offer support as a new option in the measure modal where we can insert other measure units.

### Will this feature require documentation? (Optional)

None

**Comment by AndreeaCristinaRadacina** [comment]
+1 https://secure.helpscout.net/conversation/2516986704/405425?viewId=212385

**Comment by AndreeaCristinaRadacina** [comment]
+1 https://secure.helpscout.net/conversation/2546889349/410648?viewId=212385

**Comment by Soare-Robert-Daniel** [comment]
I am unsure about this feature since Measure Input does not support distinctions like cm, m, and inch. This field type is a variant of the Number Field that can change the price. 

![image](https://github.com/user-attachments/assets/76eb3124-50c2-4930-b53a-9d15018769b9)

Adding such a feature is very complex since you might have a combination of 3 fields whose result is associated with a price. It will be more complex if you allow the fields to use different measurements in which you need to make a transformation.

**Comment by selul** [comment]
I think here this can be solved easily by adding a price multiplier, which will be applied when calculating the price. 

I.e let's say if they want to allow charging in metres but allow input of CM, they can use 0.01 as price multiplier. 


@AndreeaCristinaRadacina @Soare-Robert-Daniel what do you think?

**Comment by Soare-Robert-Daniel** [comment]
@selul, but how do you connect/correlate with the choices? (like cm, mm, m, etc).

From my point of view, each unit measure should be separated, and another selector should control its visibility.

In this case, you have a selector with the options `cm, inch` and two more inputs (one for each choice) with a custom price.

**Comment by selul** [comment]
Reviewing the tickets, I haven't seen any use cases where they use multiple units on the same product, and neither have I experienced them. Usually, if they sell based on measurements, they use the same units for all fields since the result is a square cm, m, inch, or anything. Do you have a practical example in mind where multiple units are involved? 

However, even if they use multiple units, the final price will be using just one unit i.e per square m, foot, inch, or anything, so if they have one unit in cm and one unit in inch, they will use as multiplier for one field the cm/m ratio, inch/m ratio, or whatever is the final unit of the price to convert to get the final price.

**Comment by Soare-Robert-Daniel** [comment]
The issue description is about the unit measure, so I commented on that. The linked conversations are more about combining two or more fields under one price. (Like two fields of meter to compute the prices of a square meter -- in this case, I think we need a new field input )

**Comment by selul** [comment]
> The linked conversations are more about combining two or more fields under one price. (Like two fields of meter to compute the prices of a square meter -- in this case, I think we need a new field input )

Why do you think we need a new field input? If they use two fields of meter unit and they charge on square meters even the current implementation does the job?

**Comment by Soare-Robert-Daniel** [comment]
You are right that what we have can do the job, and we can have some docs that clarify the usage.

For meters to centimeters, it is straightforward and does not need anything special (in a meter field, you can use 0.01 for centimeter). For things like inches to feet, it is not so easy; the buyer must do the math, and I think the users expect us to have this transformation so that clients do not bother. 

Like this example:

```
Let's say width is 36" which is 3 feet

Height is 24" which is 2 feet

Square feet is 6.

I need price to be 6*$5 = $30

Now if they enter say 38"x24" it rounds UP to 7 square feet and takes 7*$5 = $35
```

We can have documents that explain if you want to use both inches and feet. You should tell the user to convert or add a select field to select the unit measure and display the respect fields.

**Comment by selul** [comment]
@Soare-Robert-Daniel my proposal on having a pricing multiplier field in the field configuration would make things easier for them since if they sell in square m and they want to show the input in cm, they can use a 0.01 as multiplier, instead of asking users to input decimals in those fields. 

Same for square -> inches.

**Comment by Soare-Robert-Daniel** [comment]
In this case, yes, we can have the multiplier. But also the some docs with examples to highlight it.

**Comment by AndreeaCristinaRadacina** [comment]
@selul @Soare-Robert-Daniel I also agree with the multiplier. In the tickets I addressed, I used an extra measure input (hidden with CSS) with a default value, acting as a multiplier.

Clarifying in docs will not be an issue; I can do that once we have this done.

Thank you both for addressing this!

**Comment by selul** [comment]
updated the issue description to reflect the changes.
